spatialos-sdk/src/worker/handle.rs

```diff
@@ -0,0 +1,132 @@
+//! A type-erased [`Arc`], used to pass user-defined data types to the C API.
```

Comment:
Should this class ever be exposed to user code? It feels like implementation detail and a leaky abstraction if it is.

Currently its a `pub mod` - https://github.com/jamiebrynes7/spatialos-sdk-rs/blob/user-handle-rewrite/spatialos-sdk/src/worker/mod.rs#L9

Perhaps it better belongs in the `internal` mod?

Reply:
Ah, you're right, `UserHandle` should be entirely private to the SDK. I don't think there's any use case where exposing it to the user makes sense. I was thinking of switching it a `pub(crate)` module. Is there a strong preference between that or moving it to `internal`? Personally I'm not sure if having a separate `internal` module is necessary, though I'm not opposed to moving `handle` into `internal` if you have strong preference that way 🙂